tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env ruby
+#
+# Regenerates an animation library manifest by reading the contents of S3 bucket
+# cdo-animation-library.
+#
+# For usage, run ./rebuildAnimationLibraryManifest.rb --help
+#
+# The manifest has three important jobs:
+# 1. Provide the metadata used to render spritesheets as animations.
+# 2. Point to particular versions of the spritesheets on S3.
+# 3. Provide a precomputed search tree for quick animation searches by keyword.
+#
+# The expected output of this file is a new animationLibrary.json which is
+# consumed by the apps build.  You'll need to manually check this file into git.
+#
+# See also: The Animation Library Tech Spec (requires login):
+# https://docs.google.com/document/d/18-LVuvKd0jKTUiGo5GYReUWM5oFWCyKRyEQURJ5HCOM/edit
+#
+# TODO: Optimize: Read existing manifest and don't do full object reads for
+#       entries whose modify date hasn't changed.
```

Comment:
There are lots of TODOs in this script, and I think many of them can wait until after this PR.  I've tried to make the script user-friendly (it has a `--help` flag!) but writing shell scripts is an art and I don't have a lot of practice.  Feedback welcome!